src/heroic.rs

```diff
@@ -0,0 +1,252 @@
+            })
+        }
+    }
+
+    fn memorize_game_root(&mut self, title: &String, path: &StrictPath) {
+        let normalized = normalize_title(title);
+        if let Some(official) = self.normalized_to_official.get(&normalized) {
+            log::trace!(
+                "memorize_game_root memorizing path {} for {}",
+                path.interpret(),
+                official
+            );
+            self.games.insert(official.clone(), path.clone());
+        } else {
+            log::warn!(
+                "memorize_game_root did not find {} in manifest, no backup/restore will be done!",
+                title
+            );
+            log::trace!("memorize_game_root memorizing path {} for {}", path.interpret(), title);
+            self.games.insert(title.clone(), path.clone());
+        }
+    }
+
+    fn find_game_root(
+        &self,
+        heroic_path: String,
+        game_name: &String,
+        platform: &String,
+        game_id: &String,
+    ) -> Option<StrictPath> {
+        match platform.as_str() {
+            "windows" => {
+                // no struct for type safety used here since GamesConfig use the game id as a key name
+                let v: serde_json::Value = serde_json::from_str(
+                    &std::fs::read_to_string(format!("{}/GamesConfig/{}.json", heroic_path, game_id))
+                        .unwrap_or_default(),
+                )
+                .unwrap_or_default();
+
+                match v[&game_id]["wineVersion"]["type"].as_str().unwrap_or_default() {
+                    "wine" => {
+                        log::trace!(
+                            "find_game_root found Heroic Wine prefix for {} ({}) -> adding {}",
+                            game_name,
+                            game_id,
+                            v[&game_id]["winePrefix"].as_str().unwrap_or_default().to_string()
+                        );
+                        Some(StrictPath::new(
+                            v[&game_id]["winePrefix"].as_str().unwrap_or_default().to_string(),
+                        ))
+                    }
+
+                    "proton" => {
+                        log::trace!(
+                            "find_game_root found Heroic Proton prefix for {} ({}), adding... -> {}",
+                            game_name,
+                            game_id,
+                            format!("{}/pfx", v[&game_id]["winePrefix"].as_str().unwrap_or_default())
+                        );
+                        Some(StrictPath::new(format!(
+                            "{}/pfx",
+                            v[&game_id]["winePrefix"].as_str().unwrap_or_default()
+                        )))
+                    }
+
+                    _ => {
+                        // TODO.2022-10-07 handle unknown wine types, lutris?
+                        log::warn!(
+                            "find_game_root found Heroic Windows game {} ({}), checking... unknown wine_type: {:#?}",
```

Comment:
Oh, sorry, I misinterpreted the log message and was thinking it was something that would always log if you ran Ludusavi on Windows.

Still, even if we don't support Lutris prefixes for now, we can just document that instead of logging a warning. After the Heroic implementation is done, I can update the README's roots section with a note about what we do and don't support.

Reply:
I installed "Alba: A Wildlife Adventure" and chose a lutris variant of wine.  It uses the same structure as a plain vanilla wine, so this is not a special case to be handled.

I recommend keeping the log::warn! so that we might catch it in the logs instead of crashing ludusavi if ever GameConfig introduces a new type.  What do you think?

Reply:
Personally, I feel like we should only use warning when something has probably gone wrong. If they add a new type and we just don't support it yet, then that's a missing feature, but not a failure. If we can handle the three currently known cases, then I'd be fine with `info` for a new unknown case, though.

Reply:
reduced to a `log::info!`